### PR TITLE
feat: add rss widget

### PR DESCRIPTION
**Pull request by manuel-rw**
<br/>
<div align="center">
  
  <h3>Homarr</h3>
</div>

**Thank you for your contribution. Please ensure that your pull request meets the following pull request:**

- [ ] Builds without warnings or errors (``pnpm buid``, autofix with ``pnpm format:fix``)
- [ ] Pull request targets ``dev`` branch
- [ ] Commits follow the [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] No shorthand variable names are used (eg. ``x``, ``y``, ``i`` or any abbrevation)

image

### CODE REVIEW

File: packages/widgets/src/_inputs/widget-multi-text-input.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+  return (
+    <Combobox store={combobox}>
+      <Combobox.DropdownTarget>
+        <PillsInput
+          label={t("label")}
+          description={options.withDescription ? t("description") : undefined}
+          onClick={() => combobox.openDropdown()}
+          /* hide the error when nothing is being typed since "" is not valid but is not an explicit error */
+          error={!isValidUrl && search.length !== 0 ? "Provided URL is not valid" : undefined}
```
Comment (manuel-rw) [inline packages/widgets/src/_inputs/widget-multi-text-input.tsx:43]: Any better way of loading this translation @Meierschlumpf ?

Comment (Meierschlumpf) [reply packages/widgets/src/_inputs/widget-multi-text-input.tsx:46]: I would use the error from zod. We already translate parts of it

File: packages/api/src/router/widgets/rssFeed.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import type { RssFeed } from "@homarr/cron-jobs";
+import { createItemChannel } from "@homarr/redis";
+
+import { createOneItemMiddleware } from "../../middlewares/item";
+import { createTRPCRouter, protectedProcedure } from "../../trpc";
+
+export const rssFeedRouter = createTRPCRouter({
+  getFeeds: protectedProcedure.unstable_concat(createOneItemMiddleware("rssFeed")).query(async ({ input }) => {
```
Comment (Meierschlumpf) [inline packages/api/src/router/widgets/rssFeed.ts:8]: I think this should be public

File: packages/translation/src/lang/en.ts [hunk 3]
```diff
@@ -1021,6 +1021,21 @@ export default {
       description: "Show the current streams on your media servers",
       option: {},
     },
+    rssFeed: {
+      name: "RSS feeds",
+      description: "Monitor and display one or more generic RSS, ATOM or JSON feed",
```
Comment (Meierschlumpf) [inline packages/translation/src/lang/en.ts:1026]: should probably be plural

**Review by Meierschlumpf — CHANGES_REQUESTED** [review]

**Pull request merged**